m3/reporter.go

```diff
@@ -542,15 +541,28 @@ func (r *reporter) Tagging() bool {
 
 func (r *reporter) process() {
 	var (
-		mets  = make([]m3thrift.Metric, 0, r.freeBytes/10)
-		bytes int32
+		extraTags = sync.Pool{
+			New: func() interface{} {
+				return make([]m3thrift.MetricTag, 0, 8)
```

Comment:
shouldn't you return a ptr `make(*[]m3thrift.MetricTag, 0, 8)`, or does this work because escape analysis does not screw things up, unlike when the pool is declared outside (eg. at package level)

Reply:
Yeah, good catch. I'm almost positive this will fail escape analysis even in the current scope, because it's not returned serially after, they're returned en masse at at flush time, so as far as the compiler is concerned I can't imagine how it wouldn't escape. I'll double check. Slice pointer derefs can end up being faster anyway despite dereferencing (probably mostly due to duffcopy'ing the header).

Reply:
Looks like I was wrong; I ran a long running test and reported malloc deltas at every pool.Get and there are no allocations here. I guess it does pass the escape analysis, since it doesn't leave the outer scope of the flush loop function.